extensions/amp-experiment/1.0/mutation/attribute-mutation-default-url.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertAttributeMutationFormat} from './mutation';
+import {assertHttpsUrl} from '../../../../src/url';
+
+/**
+ * Mutation for attribute (url) mutations, for both
+ * the 'href' and 'src' attributes,
+ * on unspecifed elements.
+ *
+ * @implements {./mutation.Mutation}
+ */
+export class AttributeMutationDefaultUrl {
+  /**
+   * @param {!JsonObject} mutationRecord
+   * @param {!Array<Element>} elements
+   */
+  constructor(mutationRecord, elements) {
+    this.mutationRecord = mutationRecord;
+    this.elements = elements;
+    assertAttributeMutationFormat(mutationRecord);
+  }
+
+  /** @override */
+  validate() {
+    assertHttpsUrl(
```

Comment:
Since this is URL mutation. Can we assert that 
```
element.tagName == amp-iframe or amp-img
```

Reply:
So this would break backwards compatibility: https://github.com/ampproject/amphtml/blob/master/extensions/amp-experiment/1.0/attribute-allow-list/attribute-allow-list.js#L46

Want to confirm with @jeffjose that this wont break any partners currently using this. 😄 

That being said, We would want to ensure that the tag name does not include `amp-`, as that would require the `mutatedAttributesCallback`. Is that what you meant? 🤔

Reply:
Are you suggesting we allow changing href to all elements before? I feel we should fix that anyway even if it breaks current usage. What do you think? What's the restrictions on mutating urls?

>That being said, We would want to ensure that the tag name does not include amp-, as that would require the mutatedAttributesCallback. Is that what you meant? 🤔

I'm a bit confused 🤔... Thought we only allow mutating amp elements, with tag name starting with `amp-`?

Reply:
Oh wait, Yeah you are correct. My apologies! 😭 🙏 

I completely forgot that I did this originally but had forgotten to do the mutated attributes callback, which is why I had the origin `amp-img` mutation bug.

That being said, I'll go ahead and make the original thing you commented (Just to keep feature parity). And we can fix in the next PRs 😄

Reply:
Oh wait again haha! 😂 

So Looking at the current code the current has the following:

https://github.com/ampproject/amphtml/blob/master/extensions/amp-experiment/1.0/attribute-allow-list/attribute-allow-list.js#L75

https://github.com/ampproject/amphtml/blob/master/extensions/amp-experiment/1.0/attribute-allow-list/allowed-attribute-mutation-entry.js#L83

Meaning we match "src" and "href" for all elements. And run the `assertHttpsUrl` when we validate. Thus, if we want to keep feature parity, then this implementation is correct. But if you want to fix it in this PR we can 😄 cc @zhouyx

Reply:
Synced offline, will add it in this PR 😄